src/internet_identity/src/storage/tests.rs

```diff
@@ -130,47 +131,37 @@ fn should_serialize_subsequent_record_to_expected_memory_location() {
     for _ in 0..100 {
         storage.allocate_anchor().unwrap();
     }
-    let (anchor_number, mut anchor) = storage.allocate_anchor().unwrap();
-    assert_eq!(anchor_number, 223u64);
+    let mut anchor = storage.allocate_anchor().unwrap();
+    assert_eq!(anchor.anchor_number(), 223u64);
 
     anchor.add_device(sample_device()).unwrap();
     let expected_length = candid::encode_one(StorableAnchor::from(anchor.clone()))
         .unwrap()
         .len();
 
-    storage.write(anchor_number, anchor.clone()).unwrap();
+    storage.write(anchor.clone()).unwrap();
 
     let mut buf = vec![0u8; expected_length];
     memory.read(
         RESERVED_HEADER_BYTES + EXPECTED_RECORD_OFFSET + LENGTH_OFFSET,
         &mut buf,
     );
     let decoded_from_memory: StorableAnchor = candid::decode_one(&buf).unwrap();
-    assert_eq!(Anchor::from(decoded_from_memory), anchor);
-}
-
-#[test]
-fn should_not_write_using_anchor_number_outside_allocated_range() {
```

Comment:
Shouldn't this test remain (after adjusting)?  One can still pass an anchor with anchor_number outside of allocated range, no?

Reply:
Yeah, true. Although it is harder to do that now, as only the `storage` module itself can construct `Anchor` structs with arbitrary numbers.